lister.hh

```diff
@@ -166,10 +167,54 @@ private:
     future<directory_entry> guarantee_type(directory_entry de);
 };
 
+class queue_lister {
+    fs::path _dir;
+    lister::dir_entry_types _type;
+    lister::filter_type _filter;
+    lister::show_hidden _do_show_hidden;
+    std::unique_ptr<lister> _lister;
+    std::optional<future<>> _done;
+
+    using queue_entry_type = std::optional<directory_entry>;
```

Comment:
Why is the queue type an optional<directory_entry> and not just directory_entry?
I understand that get() will return an optional<directory_entry>, but not why every item in the queue needs to be an optional. It looks like a mistake?

Reply:
It's for pushing an in-band "end of queue" entry.
seastar::queue doesn't have an out of band "end-of-stream" concept IIUC

Reply:
Ok, somehow I missed that. pipe (which I told you is very similar to what you did :-)) is also using the same trick. I forgot it's part of pipe, not of queue.